src/EPR.RegulatorService.Facade.Core/Services/Application/ApplicationService.cs

```diff
@@ -28,7 +30,7 @@ public async Task<HttpResponseMessage> PendingApplications(Guid userId, int curr
 
         _logger.LogInformation("Attempting to fetch pending applications from the backend");
 
-        return await _httpClient.GetAsync(url);
+        return await _httpClient.GetAsync(UrlHelpers.CheckRequestURL(url));
```

Comment:
"Change this code to not construct the URL from user-controlled data" for issue
I used Uri Builder to create path and query from the input before passing in GetAsync() call which is safe and secure. I can see URL Helper method but its checking schemas and Domains not Path or query.

I did like below simple.

var uriBuilder = new UriBuilder(_httpClient.BaseAddress)
{
Path = $"decisions"
};
uriBuilder.Query = queryString;

var response = await _httpClient.GetAsync(uriBuilder.Path + uriBuilder.Query);

Do we still need URL Helper method? Let me know your thoughts..